presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileDecoderUtils.java

```diff
@@ -146,10 +146,21 @@ public static long findFirstSyncPosition(RcFileDataSource dataSource, long offse
 
             dataSource.readFully(offset + position, buffer, 0, bufferSize);
 
+            // find the starting index position of the sync sequence
             int index = bufferSlice.indexOf(sync);
-            if (index >= 0 && index + position < length) {
-                long startOfSyncSequence = offset + position + index;
-                return startOfSyncSequence;
+            if (index >= 0) {
+                // If the starting position is before the end of the search region, return the
+                // absolute start position of the sequence.
+                if (index + position < length) {
```

Comment:
We also need to explicitly check for match that crosses the bounds (i.e., we need to check the end)